genesyscloud/routing_email_route/genesyscloud_routing_email_route_proxy.go

```diff
@@ -0,0 +1,220 @@
+type getRoutingEmailRouteIdByNameFunc func(ctx context.Context, p *routingEmailRouteProxy, name string) (id string, retryable bool, respCode int, err error)
+type getRoutingEmailRouteByIdFunc func(ctx context.Context, p *routingEmailRouteProxy, domainId string, id string) (inboundRoute *platformclientv2.Inboundroute, responseCode int, err error)
+type updateRoutingEmailRouteFunc func(ctx context.Context, p *routingEmailRouteProxy, id string, domainId string, inboundRoute *platformclientv2.Inboundroute) (*platformclientv2.Inboundroute, int, error)
+type deleteRoutingEmailRouteFunc func(ctx context.Context, p *routingEmailRouteProxy, domainId string, id string) (responseCode int, err error)
+
+// routingEmailRouteProxy contains all of the methods that call genesys cloud APIs.
+type routingEmailRouteProxy struct {
+	clientConfig                     *platformclientv2.Configuration
+	routingApi                       *platformclientv2.RoutingApi
+	createRoutingEmailRouteAttr      createRoutingEmailRouteFunc
+	getAllRoutingEmailRouteAttr      getAllRoutingEmailRouteFunc
+	getRoutingEmailRouteIdByNameAttr getRoutingEmailRouteIdByNameFunc
+	getRoutingEmailRouteByIdAttr     getRoutingEmailRouteByIdFunc
+	updateRoutingEmailRouteAttr      updateRoutingEmailRouteFunc
+	deleteRoutingEmailRouteAttr      deleteRoutingEmailRouteFunc
+}
+
+// newRoutingEmailRouteProxy initializes the routing email route proxy with all of the data needed to communicate with Genesys Cloud
+func newRoutingEmailRouteProxy(clientConfig *platformclientv2.Configuration) *routingEmailRouteProxy {
+	api := platformclientv2.NewRoutingApiWithConfig(clientConfig)
+	return &routingEmailRouteProxy{
+		clientConfig:                     clientConfig,
+		routingApi:                       api,
+		createRoutingEmailRouteAttr:      createRoutingEmailRouteFn,
+		getAllRoutingEmailRouteAttr:      getAllRoutingEmailRouteFn,
+		getRoutingEmailRouteIdByNameAttr: getRoutingEmailRouteIdByNameFn,
+		getRoutingEmailRouteByIdAttr:     getRoutingEmailRouteByIdFn,
+		updateRoutingEmailRouteAttr:      updateRoutingEmailRouteFn,
+		deleteRoutingEmailRouteAttr:      deleteRoutingEmailRouteFn,
+	}
+}
+
+// getRoutingEmailRouteProxy acts as a singleton to for the internalProxy.  It also ensures
+// that we can still proxy our tests by directly setting internalProxy package variable
+func getRoutingEmailRouteProxy(clientConfig *platformclientv2.Configuration) *routingEmailRouteProxy {
+	if internalProxy == nil {
+		internalProxy = newRoutingEmailRouteProxy(clientConfig)
+	}
+	return internalProxy
+}
+
+// createRoutingEmailRoute creates a Genesys Cloud routing email route
+func (p *routingEmailRouteProxy) createRoutingEmailRoute(ctx context.Context, domainId string, routingEmailRoute *platformclientv2.Inboundroute) (*platformclientv2.Inboundroute, int, error) {
```

Comment:
Better to return the full api response instead of just the status code, we will need the full response when we start using the new diag error messages and it saves us having to change over later